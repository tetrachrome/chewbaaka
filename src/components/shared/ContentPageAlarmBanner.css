/**
 * ContentPageAlarmBanner.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 26, 2020
 * Updated  : Jul 26, 2020
 */

h4.ContentPageAlarmBannerText {
  margin: 20px auto;
  width: 60%;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 1.8rem;
  font-weight: 300;
  font-style: italic;
  text-align: center;
  line-height: 1.1;
  color: red;
  border-bottom: 3px solid black;
}
