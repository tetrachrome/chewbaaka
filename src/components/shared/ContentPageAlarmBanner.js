/**
 * ContentPageAlarmBanner.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 26, 2020
 * Updated  : Jul 26, 2020
 */

import React from 'react'

import './ContentPageAlarmBanner.css'

export default function ContentPageAlarmBanner({children}) {
  return (
    <h4 className="ContentPageAlarmBannerText">
      {children}
    </h4>
  ); 
}
