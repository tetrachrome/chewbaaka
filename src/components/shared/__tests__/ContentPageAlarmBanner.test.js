/**
 * ContentPageAlarmBanner.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Aug 17, 2020
 * Updated  : Aug 17, 2020
 */

import React from 'react'

import { render, screen } from '@testing-library/react'

import renderer from 'react-test-renderer'

import ContentPageAlarmBanner from '../ContentPageAlarmBanner'

const contentText = "This is an instance of `ContentPageAlarmBanner` component";

test('renders ContentPageAlarmBanner component', () => {
  render(
    <ContentPageAlarmBanner>
      {contentText}
    </ContentPageAlarmBanner>
  );

  const contentTextElement = screen.getByText(contentText);
  expect(contentTextElement).toBeInTheDocument();
});

test('ContentPageAlarmBanner component snapshot', () => {
  const tree = renderer
    .create(
      <ContentPageAlarmBanner>
        {contentText}
      </ContentPageAlarmBanner>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
