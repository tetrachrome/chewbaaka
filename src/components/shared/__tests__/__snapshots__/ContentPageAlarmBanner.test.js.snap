// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPageAlarmBanner component snapshot 1`] = `
<h4
  className="ContentPageAlarmBannerText"
>
  This is an instance of \`ContentPageAlarmBanner\` component
</h4>
`;
